File: src/main/java/org/example/kalkulationsprogramm/repository/MitarbeiterStundenlohnRepository.java
package org.example.kalkulationsprogramm.repository;

import org.example.kalkulationsprogramm.domain.MitarbeiterStundenlohn;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

public interface MitarbeiterStundenlohnRepository extends JpaRepository<MitarbeiterStundenlohn, Long> {

    List<MitarbeiterStundenlohn> findByMitarbeiterIdOrderByGueltigAbDesc(Long mitarbeiterId);

    Optional<MitarbeiterStundenlohn> findFirstByMitarbeiterIdAndGueltigAbLessThanEqualOrderByGueltigAbDesc(
            Long mitarbeiterId, LocalDate stichtag);
}

File: src/main/kotlin/org/example/kalkulationsprogramm/repository/MitarbeiterStundenlohnRepository.kt
package org.example.kalkulationsprogramm.repository

import java.time.LocalDate
import java.util.Optional
import org.example.kalkulationsprogramm.domain.MitarbeiterStundenlohn
import org.springframework.data.jpa.repository.JpaRepository

interface MitarbeiterStundenlohnRepository : JpaRepository<MitarbeiterStundenlohn, Long> {
    fun findByMitarbeiterIdOrderByGueltigAbDesc(mitarbeiterId: Long?): List<MitarbeiterStundenlohn>

    fun findFirstByMitarbeiterIdAndGueltigAbLessThanEqualOrderByGueltigAbDesc(
        mitarbeiterId: Long?,
        stichtag: LocalDate,
    ): Optional<MitarbeiterStundenlohn>
}
